python_code/soilIDModel_US_v3.py

```diff
@@ -819,16 +836,26 @@ def getSoilLocationBasedUS(lon, lat, plot_id):
         )
 
         # Create the list of parameters.
-        params = [
-            [ilr1_l, ilr1_r, ilr1_h],
-            [ilr2_l, ilr2_r, ilr2_h],
-            [row["dbovendry_l"], row["dbovendry_r"], row["dbovendry_h"]],
-            [row["wthirdbar_l"], row["wthirdbar_r"], row["wthirdbar_h"]],
-            [row["wfifteenbar_l"], row["wfifteenbar_r"], row["wfifteenbar_h"]],
-            [row["rfv_l"], row["rfv_r"], row["rfv_h"]],
-        ]
+        if is_constant:
+            params = [
+                [ilr1_l, ilr1_r, ilr1_h],
+                [ilr2_l, ilr2_r, ilr2_h],
+                [row["dbovendry_l"], row["dbovendry_r"], row["dbovendry_h"]],
+                [row["wthirdbar_l"], row["wthirdbar_r"], row["wthirdbar_h"]],
+                [row["wfifteenbar_l"], row["wfifteenbar_r"], row["wfifteenbar_h"]],
+            ]
+        else:
+            params = [
+                [ilr1_l, ilr1_r, ilr1_h],
+                [ilr2_l, ilr2_r, ilr2_h],
+                [row["dbovendry_l"], row["dbovendry_r"], row["dbovendry_h"]],
+                [row["wthirdbar_l"], row["wthirdbar_r"], row["wthirdbar_h"]],
+                [row["wfifteenbar_l"], row["wfifteenbar_r"], row["wfifteenbar_h"]],
+                [row["rfv_l"], row["rfv_r"], row["rfv_h"]],
+            ]
+
         # Check diagonal elements and off-diagonal range
-        if not np.all(np.diag(local_correlation_matrix) == 1) or np.any(
+        if not np.all(np.diag(local_correlation_matrix) >= 0.99999999999999) or np.any(
```

Comment:
suggestion: can you not use >= 1 here?

Reply:
In testing, I found cases where values were slightly less then 1, e.g., 0.999999999. This fixed the issue but more extensive testing is required.